lib/request.ts

```diff
@@ -377,15 +436,17 @@ export function getRequest({
 	 * 	baseUrl: 'https://api.balena-cloud.com'
 	 */
 
-	exports.refreshToken = async function ({ baseUrl }) {
+	async function refreshToken({
+		baseUrl,
+	}: Pick<BalenaRequestOptions, 'baseUrl'>): Promise<string> {
```

Comment:
There is a question to be asked here as to whether "balena-request" should know about refreshing tokens or whether that should actually be moved into "balena-auth" instead?

Reply:
I wasn't part of the discussion when this was added, but I'm guessing that the reason might be that balena-auth doesn't have access to the balena-request instance, but it's the other way around.
And since balena-request atm does refresh the token when before doing a request when the token is old enough, so we will have to think about a way that we don't have a weird circular dependency schema.